### ISSUE TITLE
[BUG][A11Y][Popover] When ui5-popover ( modal mode ) is opened, background page elements are still accessible by screen reader and can still be focused

### ISSUE THREAD

**Issue by Hubery-Shen** [SAP/ui5-webcomponents#2626]
**Dear colleagues,**
I am from SuccessFactors Web Experience Shanghai team.

**Bug Description**
**Version: Both in 1.0.0-rc.11 & 0.27.3**
When ui5-popover ( modal mode ) is opened, background page elements are still accessible by screen reader and can still be focused

![aaaaaa](https://user-images.githubusercontent.com/63704163/103521694-25c9e400-4eb4-11eb-9107-7a861f6e590a.png)

You can reproduce this issue in this example: [https://github.com/Hubery-Shen/UI5-WebComponent-Examples](https://github.com/Hubery-Shen/UI5-WebComponent-Examples)

**Expected Result**
Screen reader should only be able to read the contents of the modal dialog when its open and everything in the background should be hidden from screen readers and can not be focused.

**Comment by nnaydenow** [comment]
Hello @Hubery-Shen,

Could you please take a look on this sample to check if the issue reproducible because I'm not able to reproduce the it:
https://codesandbox.io/s/ui5-webcomponents-forked-w3tyl

Regards,
Nayden

**Comment by Milo-Shen** [comment]
Hi @nnaydenow 
Thanks you for your reply, is it convenient for you to have a call with me, i will show you the operation

**Comment by Hubery-Shen** [comment]
Hi @nnaydenow
Thanks you for your reply, is it convenient for you to have a call with me, i will show you the operation

Regards,
Hubery

**Comment by Martotko** [comment]
Hi @Hubery-Shen,

Thanks for the option. It's not convenient to have a call due to the time difference. Please just describe step by step what you are pressing so as to reproduce the issue.

Best regards,
Martin

**Comment by Hubery-Shen** [comment]
Hi @Martotko @nnaydenow 

I have prepared another demo to reproduce this problem

link is: [https://github.wdf.sap.corp/pages/xweb/goalmanagementx/app.html?mock#/goal-list?targetUserId=undefined&templateId=22&templateType=tgm](https://github.wdf.sap.corp/pages/xweb/goalmanagementx/app.html?mock#/goal-list?targetUserId=undefined&templateId=22&templateType=tgm)

Reproduce step:
1.  Click the start tour button 
2. After click the start tour button, you will see the ui-popover ( text: "Choose the tabs to switch between your performance and development goal plans" included in this ui5-popover )
3. Click the area outside the ui5-popover
4. Press the tab key
5. Then you will find i still can focus to the elements below the ui5-popover, although i have already set modal=true attribute to ui5-popover

![Screen Shot 2021-02-04 at 4 08 08 PM](https://user-images.githubusercontent.com/63704163/106866378-59906780-6707-11eb-9862-c22b7514fcae.png)

**Comment by ilhan007** [comment]
Hi, @Hubery-Shen 
@nnaydenow  and @Martotko are no longer dispatchers, @unazko is the one who is dispatching the issues this week,
so he will try to reproduce the issue whenever he can.

**Comment by Hubery-Shen** [comment]
Hi, @ilhan007 

Thanks you for you quick reply.

Thanks for your support, looking forward to your good news @unazko

**Comment by unazko** [comment]
Hello @Hubery-Shen and colleagues,

Apologies for the delay in my response.

The issue is easily reproducible in the given application:
https://github.wdf.sap.corp/pages/xweb/goalmanagementx/app.html?mock#/goal-list?targetUserId=undefined&templateId=22&templateType=tgm

Steps to reproduce as described by @Hubery-Shen.

You could also reproduce the issue in the snipped as well:
https://codesandbox.io/s/ui5-webcomponents-forked-w3tyl

Steps to reproduce:
1) Press over the "Open popover" button
2) Press over " Newsletter subscription" or "popover-content" section
3) Press out of the popover area which is grayed
4) Press tab keyboard key

Observe that the focus is moved to the first "Test" button.
This should not be possible and the screen reader will announce the button text, which is not proper behavior.

Analysis:
There is an overlay over the rest of the elements in the page managed via proper z-index.
When users presses over the overlay:
`<div class="ui5-block-layer" tabindex="1" style="z-index: 101;"></div>`
the overlay it self is set as an active element.
After that when pressing the tab keyboard key, then the next focusable elements in the page are set as active elements.

Possible solution would be to make the overlay not focusable element, or fire "focusin" event for the modal popover component, in order to return the focus back.

Could you please take over.

Best Regards,
Boyan

**Comment by Hubery-Shen** [comment]
> Hello @Hubery-Shen and @SAP/ui5-webcomponents-topic-rd,
> 
> Apologies for the delay in my response.
> 
> The issue is easily reproducible in the given application:
> https://github.wdf.sap.corp/pages/xweb/goalmanagementx/app.html?mock#/goal-list?targetUserId=undefined&templateId=22&templateType=tgm
> 
> Steps to reproduce as described by @Hubery-Shen.
> 
> You could reproduce the issue in the snipped as well:
> https://codesandbox.io/s/ui5-webcomponents-forked-w3tyl
> 
> Steps to reproduce:
> 
> 1. Press over the "Open popover" button
> 2. Press over " Newsletter subscription" or "popover-content" section
> 3. Press out of the popover area, which is greyed
> 4. Press tab keyboard key
> 
> Observe that the focus is moved to the first "Test" button.
> This should not be possible and the screen reader will announce the button text, which is not a proper behavior.
> 
> Analysis:
> There is an overlay over the rest of the elements in the page, managed via proper z-index.
> When users press over the overlay:
> `<div class="ui5-block-layer" tabindex="1" style="z-index: 101;"></div>`
> it set as an active element.
> After that when pressing the tab keyboard key, then the next focusable element in the page is set as an active element.
> 
> Possible solution would be to make the overlay not a focusable element or fire "focusin" event for the modal popover component, in order to return the focus back.
> 
> Could you please take over.
> 
> Best Regards,
> Boyan

Hi @unazko 

Thanks for your analyze
Apologies for the delay in my response.

This issue is still available in 0.29.4
Does this bug fix be released ?
